# Analyze International Debt Statistics

## 1.Overview
It's not that we humans only take debts to manage our necessities. A country may also take debt to manage its economy. For example, infrastructure spending is one costly ingredient required for a country's citizens to lead comfortable lives. The World Bank is the organization that provides debt to countries.

## 2. Objective

In this project, you are going to analyze international debt data collected by The World Bank. You are going to find the answers to questions like:
 
* What is the total amount of debt that is owed by the countries listed in the dataset?
* Which country owns the maximum amount of debt and what does that amount look like?
* What is the average amount of debt owed by countries across different debt indicators?

## 3. Data Collection

The [data](https://www.kaggle.com/datasets/theworldbank/international-debt-statistics) used in this project is provided by The World Bank. It contains both national and regional debt statistics for several countries across the globe as recorded from 1970 to 2015.
The dataset contains information about the amount of debt (in USD) owed by developing countries across several categories.


|column		   	 | type	  |
|----------------|--------|
|Country Name	 |varchar |
|Country Code	 |CHAR	  |
|Indicator Name	 |varchar |
|Indicator Code	 |varchar |
|Year			 |DECIMAL |

### 3.1. Import libraries

In [1]:
# Data manipulation
import pandas as pd
# Data visualization
import matplotlib.pyplot as plt
import seaborn as sns
# Database connection
import os
from dotenv import load_dotenv
from sqlalchemy import create_engine
from urllib.parse import quote_plus

### 3.2. Database Connection

In [2]:
load_dotenv()

# MySQL database connection using SQLAlchemy
username = os.getenv('MYSQL_ROOT_USER')
password = os.getenv('MYSQL_ROOT_PASSWORD')
host = "localhost"
port = "3306"
databasename = "PROJECT"

# URL-encode the password
encoded_password = quote_plus(password)

# Construct the connection string with the encoded password
db_uri = f"mysql+pymysql://{username}:{encoded_password}@{host}:{port}/{databasename}"
# set echo=False and all logging will be disabled
engine = create_engine(db_uri,echo=False)

In [3]:
%load_ext sql
%sql engine
%config SqlMagic.displaylimit = 20

### 3.2 Data loading
First, create a DataFrame in Python using the pandas library, and then load the dataset.

In [5]:
# Load datasets:
data = pd.read_csv('data/IDSData.csv')
print('data shape:',data.shape)
print(data.head())

data shape: (29104, 60)
  Country Name Country Code  \
0  Afghanistan          AFG   
1  Afghanistan          AFG   
2  Afghanistan          AFG   
3  Afghanistan          AFG   
4  Afghanistan          AFG   

                                      Indicator Name Indicator Code  1970  \
0  Average grace period on new external debt comm...    DT.GPA.DPPG   0.0   
1  Average grace period on new external debt comm...    DT.GPA.OFFT   0.0   
2  Average grace period on new external debt comm...    DT.GPA.PRVT   0.0   
3  Average grant element on new external debt com...    DT.GRE.DPPG   0.0   
4  Average grant element on new external debt com...    DT.GRE.OFFT   0.0   

   1971  1972  1973  1974  1975  ...  2016  2017  2018  2019  2020  2021  \
0   0.0   0.0   0.0   0.0   0.0  ...   0.0   NaN   NaN   NaN   NaN   NaN   
1   0.0   0.0   0.0   0.0   0.0  ...   0.0   NaN   NaN   NaN   NaN   NaN   
2   0.0   0.0   0.0   0.0   0.0  ...   0.0   NaN   NaN   NaN   NaN   NaN   
3   0.0   0.0   0.0   

**OBS:**
* The amount of data is almost 30k rows, we can load the data directly into the database without specifying the chunk size.
* Also, we need some preprocessing steps before inserting the data into the MySQL database.

In [6]:
## Remove all years columns, except 2016 year
data = data[['Country Name','Country Code','Indicator Name','Indicator Code','2016']]
print(data.head)

<bound method NDFrame.head of       Country Name Country Code  \
0      Afghanistan          AFG   
1      Afghanistan          AFG   
2      Afghanistan          AFG   
3      Afghanistan          AFG   
4      Afghanistan          AFG   
...            ...          ...   
29099     Zimbabwe          ZWE   
29100     Zimbabwe          ZWE   
29101     Zimbabwe          ZWE   
29102     Zimbabwe          ZWE   
29103     Zimbabwe          ZWE   

                                          Indicator Name  Indicator Code  \
0      Average grace period on new external debt comm...     DT.GPA.DPPG   
1      Average grace period on new external debt comm...     DT.GPA.OFFT   
2      Average grace period on new external debt comm...     DT.GPA.PRVT   
3      Average grant element on new external debt com...     DT.GRE.DPPG   
4      Average grant element on new external debt com...     DT.GRE.OFFT   
...                                                  ...             ...   
29099            

In [7]:
# We need to remove the following "country name" to avoid misunderstandings during further analysis.
countries_to_remove = ['East Asia & Pacific (excluding high income)',
                       'Europe & Central Asia (excluding high income)',
                       'Latin America & Caribbean (excluding high income)',
                       'Least developed countries: UN classification',
                       'Low & middle income',
                       'Low income',
                       'Lower middle income',
                       'Middle East & North Africa (excluding high income)',
                       'Middle income',
                       'South Asia',
                       'Sub-Saharan Africa (excluding high income)',
                       'Upper middle income']

data = data[~data['Country Name'].isin(countries_to_remove)]

In [8]:
# Before to create table in MySQL and populate the database, we need to rename the columns.
data.rename(columns={'Country Name':'country_name',
                     'Country Code':'country_code',
                     'Indicator Name':'indicator_name',
                     'Indicator Code':'indicator_code',
                     '2016':'debt'}, inplace=True)
data.head()

country_name country_code  \
0  Afghanistan          AFG   
1  Afghanistan          AFG   
2  Afghanistan          AFG   
3  Afghanistan          AFG   
4  Afghanistan          AFG   

                                      indicator_name indicator_code  debt  
0  Average grace period on new external debt comm...    DT.GPA.DPPG   0.0  
1  Average grace period on new external debt comm...    DT.GPA.OFFT   0.0  
2  Average grace period on new external debt comm...    DT.GPA.PRVT   0.0  
3  Average grant element on new external debt com...    DT.GRE.DPPG   0.0  
4  Average grant element on new external debt com...    DT.GRE.OFFT   0.0

In [9]:
# Consider only Debt (DT) in "Indicator Code" column, the rest must be removed:
# For example: 
# not remove:DT.GPA.DPPG
# remove: FI.RES.TOTL.MO, BX.KLT.DINV.CD.DT, NY.GNP.MKTP.CD
data = data[data['indicator_code'].str.startswith('DT.')]
data.head()

country_name country_code  \
0  Afghanistan          AFG   
1  Afghanistan          AFG   
2  Afghanistan          AFG   
3  Afghanistan          AFG   
4  Afghanistan          AFG   

                                      indicator_name indicator_code  debt  
0  Average grace period on new external debt comm...    DT.GPA.DPPG   0.0  
1  Average grace period on new external debt comm...    DT.GPA.OFFT   0.0  
2  Average grace period on new external debt comm...    DT.GPA.PRVT   0.0  
3  Average grant element on new external debt com...    DT.GRE.DPPG   0.0  
4  Average grant element on new external debt com...    DT.GRE.OFFT   0.0

Second, verify if the `international_debt` table exists in database. Create `international_debt` table in MySQL Database:

In [10]:
%%sql
-- Check 'international_debt' table, if exist:
SHOW TABLES LIKE 'international_debt';

Running query in 'mysql+pymysql://root:***@localhost:3306/PROJECT'

1 rows affected.

+----------------------------------------+
| Tables_in_PROJECT (international_debt) |
+----------------------------------------+
|           international_debt           |
+----------------------------------------+

In [11]:
%%sql
-- Remove 'international_debt' TABLE, if exist:
DROP TABLE IF EXISTS international_debt;

Running query in 'mysql+pymysql://root:***@localhost:3306/PROJECT'

++
||
++
++

In [12]:
%%sql
-- Create 'international_debt' table:
CREATE TABLE international_debt(
    country_name VARCHAR(64),
    country_code CHAR(3),
    indicator_name VARCHAR(100),
    indicator_code VARCHAR(32),
    debt DECIMAL(18,2)
);

Running query in 'mysql+pymysql://root:***@localhost:3306/PROJECT'

++
||
++
++

Finally, populate `international_debt` table from the python dataframe.

In [13]:
data.to_sql(name = "international_debt", 
            con = engine,
            if_exists = 'append',
            index= False)

engine.dispose()

In [14]:
%%sql
-- Check 'international_debt' table:
SELECT * FROM international_debt LIMIT 5;

Running query in 'mysql+pymysql://root:***@localhost:3306/PROJECT'

5 rows affected.

+--------------+--------------+-------------------------------------------------------------------------+----------------+------+
| country_name | country_code |                              indicator_name                             | indicator_code | debt |
+--------------+--------------+-------------------------------------------------------------------------+----------------+------+
| Afghanistan  |     AFG      |      Average grace period on new external debt commitments (years)      |  DT.GPA.DPPG   | 0.00 |
| Afghanistan  |     AFG      | Average grace period on new external debt commitments, official (years) |  DT.GPA.OFFT   | 0.00 |
| Afghanistan  |     AFG      |  Average grace period on new external debt commitments, private (years) |  DT.GPA.PRVT   | 0.00 |
| Afghanistan  |     AFG      |        Average grant element on new external debt commitments (%)       |  DT.GRE.DPPG   | 0.00 |
| Afghanistan  |     AFG      |   Average grant element on new external debt commitments, official (%)  |  DT.GRE.OFFT   | 0.00 |
+--------------+--------------+-------------------------------------------------------------------------+----------------+------+

## 4.Exploratory Data Analysis(EDA):
### 4.1. Data Dimensions

In [15]:
%%sql
SELECT COUNT(*) AS Total_rows FROM international_debt;

Running query in 'mysql+pymysql://root:***@localhost:3306/PROJECT'

1 rows affected.

+------------+
| Total_rows |
+------------+
|   24924    |
+------------+

**OBS:** There are almost 25K rows.

### 4.2. Data Type

In [16]:
%%sql
DESCRIBE international_debt;

Running query in 'mysql+pymysql://root:***@localhost:3306/PROJECT'

5 rows affected.

+----------------+---------------+------+-----+---------+-------+
|     Field      |      Type     | Null | Key | Default | Extra |
+----------------+---------------+------+-----+---------+-------+
|  country_name  |  varchar(64)  | YES  |     |   None  |       |
|  country_code  |    char(3)    | YES  |     |   None  |       |
| indicator_name |  varchar(100) | YES  |     |   None  |       |
| indicator_code |  varchar(32)  | YES  |     |   None  |       |
|      debt      | decimal(18,2) | YES  |     |   None  |       |
+----------------+---------------+------+-----+---------+-------+

**OBS**: Data type is ok.

### 4.3. Missing values
Let's identify missing values to explore the limitations of our database.


In [17]:
%%sql
-- For Numeric, date and time Data Types: missing value = NULL
-- For String: missing value = NULL or ''
SELECT
    COUNT(CASE WHEN country_name IS NULL OR country_name = '' THEN 1 END) AS m_country_name,
    COUNT(CASE WHEN country_code IS NULL OR country_code = '' THEN 1 END) AS m_country_code,
    COUNT(CASE WHEN indicator_name IS NULL OR indicator_name = '' THEN 1 END) AS m_indicator_name,
    COUNT(CASE WHEN debt IS NULL THEN 1 END) AS m_debt
FROM international_debt;

Running query in 'mysql+pymysql://root:***@localhost:3306/PROJECT'

1 rows affected.

+----------------+----------------+------------------+--------+
| m_country_name | m_country_code | m_indicator_name | m_debt |
+----------------+----------------+------------------+--------+
|       0        |       0        |        0         |   88   |
+----------------+----------------+------------------+--------+

**OBS:** There are 88 missing values in `debt` column, which only makes up 0.35% of all rows (24924) in the database.

### 4.4. Duplicated rows:

In [18]:
%%sql
    
SELECT *, COUNT(*)
FROM international_debt
GROUP BY country_name, country_code,
         indicator_name, indicator_code, debt
HAVING COUNT(*) > 1;

Running query in 'mysql+pymysql://root:***@localhost:3306/PROJECT'

+--------------+--------------+----------------+----------------+------+----------+
| country_name | country_code | indicator_name | indicator_code | debt | COUNT(*) |
+--------------+--------------+----------------+----------------+------+----------+
+--------------+--------------+----------------+----------------+------+----------+

**OBS:** There are not duplicate rows.

## 5.Data Preprocessing:


## 6. Data Analysis


### 6.1. Finding the number of distinct countries
From the first ten rows, we can see the amount of debt owed by Afghanistan in the different debt indicators. But we do not know the number of different countries we have on the table. There are repetitions in the country names because a country is most likely to have debt in more than one debt indicator.

Without a count of unique countries, we will not be able to perform our statistical analyses holistically. In this section, we are going to extract the number of unique countries present in the table.

In [19]:
%%sql
SELECT 
    COUNT(DISTINCT(country_name)) AS total_distinct_countries
FROM international_debt;

Running query in 'mysql+pymysql://root:***@localhost:3306/PROJECT'

1 rows affected.

+--------------------------+
| total_distinct_countries |
+--------------------------+
|           124            |
+--------------------------+

**OBS:** We can see there are a total of 124 countries present on the table. 

### 6.2. Finding out the distinct debt indicators
As we saw in the first section, there is a column called `indicator_name` that briefly specifies the purpose of taking the debt. Just beside that column, there is another column called `indicator_code` which symbolizes the category of these debts. Knowing about these various debt indicators will help us to understand the areas in which a country can possibly be indebted to.

In [20]:
%%sql
SELECT 
    DISTINCT(indicator_code) AS distinct_debt_indicators,
    indicator_name
FROM international_debt
GROUP BY distinct_debt_indicators, indicator_name
ORDER BY distinct_debt_indicators ASC
LIMIT 10;

Running query in 'mysql+pymysql://root:***@localhost:3306/PROJECT'

10 rows affected.

+--------------------------+------------------------------------------------------------------------------------------------+
| distinct_debt_indicators |                                         indicator_name                                         |
+--------------------------+------------------------------------------------------------------------------------------------+
|      DT.AMT.BLAT.CD      |                               PPG, bilateral (AMT, current US$)                                |
|      DT.AMT.BLTC.CD      |                         PPG, bilateral concessional (AMT, current US$)                         |
|      DT.AMT.DIMF.CD      |                               IMF repurchases (AMT, current US$)                               |
|      DT.AMT.DLTF.CD      |           Principal repayments on external debt, long-term + IMF (AMT, current US$)            |
|      DT.AMT.DLXF.CD      |              Principal repayments on external debt, long-term (AMT, current US$)               |
|      DT.AMT.DPNG.CD      |     Principal repayments on external debt, private nonguaranteed (PNG) (AMT, current US$)      |
|      DT.AMT.DPPG.CD      | Principal repayments on external debt, public and publicly guaranteed (PPG) (AMT, current US$) |
|      DT.AMT.MIBR.CD      |                                  PPG, IBRD (AMT, current US$)                                  |
|      DT.AMT.MIDA.CD      |                                  PPG, IDA (AMT, current US$)                                   |
|      DT.AMT.MLAT.CD      |                              PPG, multilateral (AMT, current US$)                              |
+--------------------------+------------------------------------------------------------------------------------------------+

### 6.3. Totaling the amount of debt owed by the countries
As mentioned earlier, the financial debt of a particular country represents its economic state. But if we were to project this on an overall global scale, how will we approach it?

Let's switch gears from the debt indicators now and find out the total amount of debt (in USD) that is owed by the different countries. This will give us a sense of how the overall economy of the entire world is holding up.

In [21]:
%%sql
-- Add up all the world debt and divide it by 1 million
SELECT 
    ROUND(SUM(debt)/1000000, 2) as total_debt
FROM international_debt;

Running query in 'mysql+pymysql://root:***@localhost:3306/PROJECT'

1 rows affected.

+-------------+
|  total_debt |
+-------------+
| 57945019.05 |
+-------------+

**OBS:** The total debt is almost 57'945'019.05 Million USD, or 57.95 Trillion USD.

### 6.4. Country with the highest debt


Now that we have the exact total of the amounts of debt owed by several countries, let's now find out the country that owns the highest amount of debt along with the amount. Note that this debt is the sum of different debts owed by a country across several categories. This will help to understand more about the country in terms of its socio-economic scenarios. We can also find out the category in which the country owns its highest debt. But we will leave that for now.

In [23]:
%%sql
-- Add up all the world debt and divide it by 1 million
-- Group by country, Sort descending and filter by the first result
SELECT 
    country_name, 
    ROUND(SUM(debt)/1000000, 2) as total_debt
FROM international_debt
GROUP BY country_name
ORDER BY total_debt DESC 
LIMIT 1;

Running query in 'mysql+pymysql://root:***@localhost:3306/PROJECT'

1 rows affected.

+--------------+------------+
| country_name | total_debt |
+--------------+------------+
|    China     | 8039991.06 |
+--------------+------------+

**OBS:** China is the country with the highest debt (8 Trillion USD).

### 6.5. Average amount of debt across indicators


We now have a brief overview of the dataset and a few of its summary statistics. We already have an idea of the different debt indicators in which the countries owe their debts. We can dig even further to find out on an average how much debt a country owes? This will give us a better sense of the distribution of the amount of debt across different indicators.

In [24]:
%%sql
-- Add up all the world debt and divide it by 1 million, rename as average_debt
-- Include indicator_name and indicator_code as debt_indicator.
-- group by debt_indicator and indicator_name
-- Order by average_debt, Sort descending and filter by the Top-10 results.
SELECT 
    indicator_code AS debt_indicator,
    ROUND(AVG(debt)/1000000,2) as average_debt,
    indicator_name
FROM international_debt
GROUP BY debt_indicator, indicator_name
ORDER BY average_debt DESC
LIMIT 10;

Running query in 'mysql+pymysql://root:***@localhost:3306/PROJECT'

10 rows affected.

+----------------+--------------+-------------------------------------------------------------------------------+
| debt_indicator | average_debt |                                 indicator_name                                |
+----------------+--------------+-------------------------------------------------------------------------------+
| DT.DOD.DECT.CD |   57922.82   |                 External debt stocks, total (DOD, current US$)                |
| DT.DOD.DLXF.CD |   42814.98   |               External debt stocks, long-term (DOD, current US$)              |
| DT.DOD.VTOT.CD |   27639.65   |             External debt stocks, variable rate (DOD, current US$)            |
| DT.DOD.PVLX.CD |   22747.41   |                  Present value of external debt (current US$)                 |
| DT.DOD.DPPG.CD |   22173.14   | External debt stocks, public and publicly guaranteed (PPG) (DOD, current US$) |
| DT.DOD.PUBS.CD |   22006.05   |        External debt stocks, long-term public sector (DOD, current US$)       |
| DT.DOD.PRVS.CD |   20808.93   |       External debt stocks, long-term private sector (DOD, current US$)       |
| DT.DOD.DPNG.CD |   20641.84   |      External debt stocks, private nonguaranteed (PNG) (DOD, current US$)     |
| DT.DOD.PNGC.CD |   17027.52   |          PNG, commercial banks and other creditors (DOD, current US$)         |
| DT.DOD.DSTC.CD |   14047.12   |              External debt stocks, short-term (DOD, current US$)              |
+----------------+--------------+-------------------------------------------------------------------------------+

**OBS:** We can see that the indicator `DT.DOD.DECT.CD` tops the chart of average debt. This category includes External debt stocks. Total external debt is debt owed to nonresidents repayable in currency, goods, or services. More information about this category can be found [here](https://data.worldbank.org/indicator/DT.DOD.DECT.CD).

### 6.6. The highest amount of principal repayments
We can investigate this a bit more so as to find out which country owes the highest amount of debt in the category of long term debts (DT.AMT.DLXF.CD). Since not all the countries suffer from the same kind of economic disturbances, this finding will allow us to understand that particular country's economic condition a bit more specifically.

In [25]:
%%sql
-- Add up all the world debt and divide it by 1 million, rename as average_debt.
-- Include and group by country_name, indicator_name and indicator_code.
-- Filter by indicator_code = 'DT.AMT.DLXF.CD'
-- Order by average_debt, Sort descending and filter by the Top-10 results.
SELECT 
    country_name, 
    ROUND(AVG(debt)/1000000,2) AS average_debt,
    indicator_name,
    indicator_code
FROM international_debt
GROUP BY country_name, indicator_name, indicator_code
HAVING indicator_code = 'DT.AMT.DLXF.CD'
ORDER BY average_debt DESC
LIMIT 10;

Running query in 'mysql+pymysql://root:***@localhost:3306/PROJECT'

10 rows affected.

+--------------------+--------------+---------------------------------------------------------------------+----------------+
|    country_name    | average_debt |                            indicator_name                           | indicator_code |
+--------------------+--------------+---------------------------------------------------------------------+----------------+
|       Brazil       |   94709.39   | Principal repayments on external debt, long-term (AMT, current US$) | DT.AMT.DLXF.CD |
|       China        |   91041.78   | Principal repayments on external debt, long-term (AMT, current US$) | DT.AMT.DLXF.CD |
|       India        |   65971.74   | Principal repayments on external debt, long-term (AMT, current US$) | DT.AMT.DLXF.CD |
|       Turkey       |   61586.52   | Principal repayments on external debt, long-term (AMT, current US$) | DT.AMT.DLXF.CD |
|       Mexico       |   59129.58   | Principal repayments on external debt, long-term (AMT, current US$) | DT.AMT.DLXF.CD |
|     Indonesia      |   57213.15   | Principal repayments on external debt, long-term (AMT, current US$) | DT.AMT.DLXF.CD |
| Russian Federation |   51555.33   | Principal repayments on external debt, long-term (AMT, current US$) | DT.AMT.DLXF.CD |
|     Kazakhstan     |   17412.88   | Principal repayments on external debt, long-term (AMT, current US$) | DT.AMT.DLXF.CD |
|      Romania       |   16630.36   | Principal repayments on external debt, long-term (AMT, current US$) | DT.AMT.DLXF.CD |
|      Thailand      |   12332.72   | Principal repayments on external debt, long-term (AMT, current US$) | DT.AMT.DLXF.CD |
+--------------------+--------------+---------------------------------------------------------------------+----------------+

**OBS:** Brazil and China has the highest amount of debt in the long-term debt (DT.AMT.DLXF.CD) category. This is verified by The World Bank. It is often a good idea to verify our analyses like this since it validates that our investigations are correct.

### 6.7. The most common debt indicator
We saw that `External debt stocks, total` is the top category when it comes to the average amount of debt. But is it the most common indicator in which the countries owe their debt? Let's find that out.

In [26]:
%%sql
SELECT 
    indicator_name,
    COUNT(indicator_code) AS indicator_count
FROM international_debt
GROUP BY indicator_code, indicator_name
ORDER BY indicator_count DESC
LIMIT 10;

Running query in 'mysql+pymysql://root:***@localhost:3306/PROJECT'

10 rows affected.

+-------------------------------------------------------------------------+-----------------+
|                              indicator_name                             | indicator_count |
+-------------------------------------------------------------------------+-----------------+
|      Average grace period on new external debt commitments (years)      |       124       |
| Average grace period on new external debt commitments, official (years) |       124       |
|  Average grace period on new external debt commitments, private (years) |       124       |
|        Average grant element on new external debt commitments (%)       |       124       |
|   Average grant element on new external debt commitments, official (%)  |       124       |
|   Average grant element on new external debt commitments, private (%)   |       124       |
|          Average interest on new external debt commitments (%)          |       124       |
|     Average interest on new external debt commitments, official (%)     |       124       |
|      Average interest on new external debt commitments, private (%)     |       124       |
|        Average maturity on new external debt commitments (years)        |       124       |
+-------------------------------------------------------------------------+-----------------+

**OBS:** There are many debt indicators in which all the countries listed in our dataset have taken debt.

### 6.8. Other viable debt issues and conclusion
Let's change tracks from debt_indicators now and focus on the amount of debt again. Let's find out the maximum amount of debt across the indicators along with the respective country names. With this, we will be in a position to identify the other plausible economic issues a country might be going through.

In [27]:
%%sql
SELECT
    country_name,
    indicator_name,
    MAX(debt) as maximum_debt
FROM international_debt
GROUP BY country_name, indicator_name
ORDER BY maximum_debt DESC
LIMIT 15;

Running query in 'mysql+pymysql://root:***@localhost:3306/PROJECT'

15 rows affected.

+--------------------+----------------------------------------------------------------------+------------------+
|    country_name    |                            indicator_name                            |   maximum_debt   |
+--------------------+----------------------------------------------------------------------+------------------+
|       China        |            External debt stocks, total (DOD, current US$)            | 1429467967000.00 |
|       China        |         External debt stocks, short-term (DOD, current US$)          | 801396078000.00  |
|       China        |          External debt stocks, long-term (DOD, current US$)          | 618675469000.00  |
|       Brazil       |            External debt stocks, total (DOD, current US$)            | 543257252000.00  |
|       China        |        External debt stocks, variable rate (DOD, current US$)        | 525975135000.00  |
| Russian Federation |            External debt stocks, total (DOD, current US$)            | 524685938000.00  |
|       Brazil       |          External debt stocks, long-term (DOD, current US$)          | 482174315000.00  |
| Russian Federation |          External debt stocks, long-term (DOD, current US$)          | 471961163000.00  |
|       China        | External debt stocks, private nonguaranteed (PNG) (DOD, current US$) | 460000146000.00  |
|       China        |  External debt stocks, long-term private sector (DOD, current US$)   | 459447764000.00  |
|       India        |            External debt stocks, total (DOD, current US$)            | 456139934000.00  |
|       Mexico       |            External debt stocks, total (DOD, current US$)            | 422656826000.00  |
|       Turkey       |            External debt stocks, total (DOD, current US$)            | 405656066000.00  |
|       Brazil       |        External debt stocks, variable rate (DOD, current US$)        | 373236999000.00  |
|       India        |          External debt stocks, long-term (DOD, current US$)          | 366970842000.00  |
+--------------------+----------------------------------------------------------------------+------------------+

## 7.Conclusion

* The database has 124 countries, numerous debt indicator categories, and a total debt of 57,945,019.05 million USD (57.95 trillion USD).
* The country with the highest debt is `China`, with 8,039,991.06 million USD (8.04 trillion USD). This is the price to pay for accelerated economic growth.
* The `external debt stocks, total` indicator (DT.DOD.DECT.CD), tops the chart for average debt.
* Brazil and China have the highest amount of debt in the `long-term debt` (DT.AMT.DLXF.CD) indicator.

## 8.References
* https://www.datacamp.com/projects/754
* https://www.kaggle.com/datasets/theworldbank/international-debt-statistics
* https://github.com/Doj-i/SQL-Analyze-International-Debt-Statistics/blob/master/notebook.ipynb
* https://datatopics.worldbank.org/debt/ids/country/CHN